autogen/tests/test.bats

```diff
@@ -0,0 +1,11 @@
+TEST_PKG_VERSION="$(echo "${TEST_PKG_IDENT}" | cut -d/ -f3)"
+
+@test "Autogen version command returns expected version" {
+  result="$(hab pkg exec "${TEST_PKG_IDENT}" autogen --version)"
+  [[ "$result" =~ $TEST_PKG_VERSION  ]]
```

Comment:
https://github.com/habitat-sh/core-plans/blob/master/docs/dev/policy_documents/standards-for-tests.md needs to be updated because its examples look like this, but it's important to never use `[[` with BATs. Some context: https://github.com/sstephenson/bats/issues/49

The pattern we're using (that we'll update the doc to) is to have the result cut/sed/awk the command result down to only the version term; then having the test line be `[ "$result" = $TEST_PKG_VERSION  ]`

Reply:
Raised https://github.com/habitat-sh/core-plans/issues/2716 for this